source/Ship.cpp

```diff
@@ -3550,10 +3554,12 @@ int Ship::TakeDamage(vector<Visual> &visuals, const DamageDealt &damage, const G
 	else if(heat < .9 * MaximumHeat())
 		isOverheated = false;
 
+	double ionWeaponJamChance = damage.Ion() ? CalculateJamChance(Energy(), ionization) : 0.;
```

Comment:
Why only check the jam change if the weapon that the ship is taking damage from deals ion damage? I feel like either the ion damage taken shouldn't matter, or all damage types that are checked against should be taken into account. For example:
```c++
bool enraged = false;
enraged |= damage.Shield() && Shields() < .9;
enraged |= damage.Hull() && Hull() < .9;
enraged |= damage.Heat() && isOverheated;
enraged |= damage.Ion() && CalculateJamChance(Energy(), ionization) > 0.1;
enraged |= damage.Slowing() && slowness > 10;
if(sourceGovernment && !sourceGovernment->IsEnemy(government)
		&& (!personality.IsForbearing() || enraged)
```